addons/godottpd/http_server.gd

```diff
@@ -143,4 +167,43 @@ func _perform_current_request(client: StreamPeer, request: HttpRequest):
 					found = true
 					router.router.handle_options(request, response)
 	if not found:
+		if not request.path.get_extension() in ["gd"]:
```

Comment:
I guess this is the point where local file serving is handled, right?

I'd like to suggest, that we maybe handle this using a special router instead, so that the developer has a handy tool to serve files but can control where to enable and disable it.

Reply:
Ok I understand what you are talking about.
The thing is that files are served only if the specific extension of the file is expressed.
Ex.
```
GET /asset/index
```
this will first try to serve the HttpRouter, if defined for the "/asset/index" and "/asset/index/" path, then will 404 if the router does not exist. No files will ever be served since there's no extension
```
GET /asset/index.html
```
this will (1) try to serve the HttpRouter for "/asset/index.html" and "/asset/index.html/" if it is defined, then (2) will try to serve the "index.html" file in "res://src"+"/asset/index.html" if exists and it is accessible, then (3) will 404.
What I'm saying is that, if the file extension is not defined, files will never be served, and even if the extension is defined but there is an HttpRouter defined with the same extension, the Router will be served.
I think that's pretty elastic, since users don't have to define a Router for each file they want to serve, but if they want to serve a Router instead of a file they only have to "overwrite" the route to the file (name+extension).

Otherwise we could implement a special character to identify a File serving instead of a Router one, something like `/assets/§index`. But this will need to add more logic to prevent a Route serving, and also there's no way to define what file should be served (for example if there are "index.html" and "index.css" in the same folder), unless the path requested is `/assets/$index.html`.
Do you have any other idea?

Reply:
An alternative could be how non-js frameworks like Spring Boot do, providing an HTML Template to the Route.
We could make something like 
```
var path: String = "/page/index"
var router: HttpRouter = HttpRouter.new()
var model: Model = Model.new()
server.route(path, router, model)
```
where the `Model` class is an HTML Parser, which reads text files and parses them as string.
If "Model" is provided in the .route() function, calling the path will directly send the file content using Model attributes.
This could also let us implement some features, like defining placeholders in HTML Templates and replace them with attributes passed to the Model class

Reply:
I'll go with your suggestion and implement the StaticFilesRouter (should we call this the same way? I was thinking of `HttpFileRouter` since you named yours `HttpRouter`).

Also I think that the HTML Parser I was talking about could consist of its own plugin, making something very similar to how [Angular handles variables in HTML Templates](https://angular.io/guide/template-reference-variables#using-ngform-with-template-variables), or more similarly to [Thmeleaf](https://www.thymeleaf.org/doc/tutorials/2.1/thymeleafspring.html#creating-a-form).